server/etcdserver/corrupt.go

```diff
@@ -510,6 +521,10 @@ func (h *hashKVHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, "bad path", http.StatusBadRequest)
 		return
 	}
+	if gcid := r.Header.Get("X-Etcd-Cluster-ID"); gcid != h.server.cluster.ID().String() {
+		http.Error(w, rpctypes.ErrGRPCClusterIdMismatch.Error(), http.StatusPreconditionFailed)
```

Comment:
```suggestion
		http.Error(w, mvcc.ErrClusterIdMismatch.Error(), http.StatusPreconditionFailed)
```

Reply:
Or expose errClusterIDMismatch, and use it directly https://github.com/etcd-io/etcd/blob/693f25d6b98f5070527bc2b461f0992227c01c14/server/etcdserver/api/rafthttp/http.go#L57